src/fcntl.rs

```diff
@@ -180,33 +180,88 @@ pub fn renameat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(old_dirfd: Option<Ra
     Errno::result(res).map(drop)
 }
 
-fn wrap_readlink_result(v: &mut Vec<u8>, res: ssize_t) -> Result<OsString> {
+fn wrap_readlink_result(mut v: Vec<u8>, res: ssize_t) -> Result<OsString> {
     match Errno::result(res) {
         Err(err) => Err(err),
```

Comment:
This line is dead code.  We only ever call `wrap_readlink_result` when `res` is >= 0, so you may as well eliminate the `Errno::result` part.  I think that issued predated your changes, but since you're here would you please fix it?